Generate Microsoft.PolicyInsights SDK from latest swagger

eladperets opened the pull request:
- Added policy metadata (new resource type) tests
- Update existing tests for new API versions (remediation and policy states)

Swagger PRs:
https://github.com/Azure/azure-rest-api-specs/pull/7675
https://github.com/Azure/azure-rest-api-specs/pull/7717
https://github.com/Azure/azure-rest-api-specs/pull/7505

eladperets committed:
Generate Microsoft.PolicyInsights SDK from latest swagger

- Added policy metadata (new resource type) tests
- Update existing tests for new API versions (remediation and policy states)

eladperets committed:
Add policy metadata to the API info

eladperets commented on sdk/policyinsights/Microsoft.Azure.Management.PolicyInsights/src/Generated/SdkInfo_PolicyInsightsClient.cs:
```diff
@@ -24,6 +24,7 @@ internal static partial class SdkInfo
                 new Tuple<string, string, string>("PolicyInsights", "PolicyStates", "2019-10-01"),
                 new Tuple<string, string, string>("PolicyInsights", "PolicyTrackedResources", "2018-07-01-preview"),
                 new Tuple<string, string, string>("PolicyInsights", "Remediations", "2019-07-01"),
+                new Tuple<string, string, string>("PolicyInsights", "PolicyMetadata", "2019-10-01"),
```
The header of this file says it's autogenerated, but it wasn't part of the autorest output and we updated it manually. Is it still needed?

isra-fel replied:
Hi Elad, please try [updating build tools](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#updating-build-tools)

eladperets replied:
Thanks!

eladperets committed:
small fix

isra-fel reviewed (changes requested):
Hi @eladperets , sorry for the delay. Could you
- Add a link to the swagger spec review PR in the description
- Include the `.txt` file generated when you run `generate.ps1`[(details)](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec)
- Update api tag in `AzSdk.RP.props`, release notes in csproj
- Update versions in csproj and `assemblyinfo.cs` if you are going to release

Thanks

eladperets committed:
Merge branch 'master' of https://github.com/Azure/azure-sdk-for-net

eladperets committed:
Address PR comments

- Incremented the major version because the new SDK removed a call that
was available in previous versions

eladperets commented:
> Hi @eladperets , sorry for the delay. Could you
> 
> * Add a link to the swagger spec review PR in the description
> * Include the `.txt` file generated when you run `generate.ps1`[(details)](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/adx_netsdk_process.md#sdk-generation-from-updated-spec)
> * Update api tag in `AzSdk.RP.props`, release notes in csproj
> * Update versions in csproj and `assemblyinfo.cs` if you are going to release
> 
> Thanks

Done
